test/functional/p2p_disconnect_ban.py

```diff
@@ -85,9 +86,9 @@ def run_test(self):
         self.log.info("setban: test banning with absolute timestamp")
         self.nodes[1].setban("192.168.0.2", "add", old_time + 120, True)
 
-        # Move time forward by 3 seconds so the third ban has expired
+        # Move time forward by 3 seconds so the fourth ban has expired
         self.nodes[1].setmocktime(old_time + 3)
-        assert_equal(len(self.nodes[1].listbanned()), 4)
+        assert_equal(len(self.nodes[1].listbanned()), 5)
 
         self.log.info("Test ban_duration and time_remaining")
         for ban in self.nodes[1].listbanned():
```

Comment:
Looking good!  What if we button things up by adding checks for the onion address inside the for loop `for ban in self.nodes[1].listbanned():`?

https://github.com/bitcoin/bitcoin/blob/7cdef4bc1148a6641ad6cfb2ade90cbe8fba82cb/test/functional/p2p_disconnect_ban.py#L93-L103

Reply:
Nice idea, done!